sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -118,6 +118,7 @@ RT::PiKernel ProgramManager::getOrCreateKernel(OSModuleHandle M,
   RT::PiKernel &Kernel = KernelsCache[KernelName];
   if (!Kernel) {
     PI_CALL(RT::piKernelCreate, Program, KernelName.c_str(), &Kernel);
+    // TODO validate the result, check for errors
```

Comment:
I thought PI_CALL checked for errors. Or did you mean something else?

Reply:
AFAIK, PI_CALL just throws sycl::runtime_error with no useful for the user information, so we should write more fine grained error checking to improve user experience.
But I agree the comment is a little bit unclear.